src/endpoints/data_objects/impl.cpp

```diff
@@ -300,6 +319,22 @@ namespace
                 }
 
                 auto conn = irods::get_connection(client_info->username);
+
+                // Enable ticket if the request includes one.
+                if (const auto iter = _args.find("ticket"); iter != std::end(_args)) {
+                    if (const auto ec = irods::enable_ticket(conn, iter->second); ec < 0) {
+                        res.result(http::status::internal_server_error);
+                        res.body() = json{
+                            {"irods_response", {
+                                {"error_code", ec},
+                                {"error_message", "Error enabling ticket on connection."}
+                            }}
+                        }.dump();
+                        res.prepare_payload();
+                        return _sess_ptr->send(std::move(res));
+                    }
+                }
+
```

Comment:
this block of code is repeated... looks like 7-8x so far... any sane way to unify?

Reply:
Yes, there is. I'm going to take care of that soon.

Reply:
ah, very nice.

Reply:
The more I look at these, the more I feel it is fine.

The only lines that seem worthy of unifying are the ones dealing with constructing an error response.

If we try and encapsulate the inner most if-statement, we either have to throw an exception which isn't ideal (since we can just return) or we have to return more info from the new function.

I'm going to consolidate the body of the inner if-statement.

Reply:
Created issue #85 for this.

Leaving as is for now.